packages/cosmwasm/src/cw3cosmwasmclient.ts

```diff
@@ -0,0 +1,197 @@
+}
+
+export class Cw3CosmWasmClient extends SigningCosmWasmClient {
+  private readonly cw3ContractAddress: string;
+
+  public constructor(
+    apiUrl: string,
+    senderAddress: string,
+    signer: OfflineSigner,
+    cw3ContractAddress: string,
+    gasPrice?: GasPrice,
+    gasLimits?: Partial<GasLimits<CosmWasmFeeTable>>,
+    broadcastMode?: BroadcastMode,
+  ) {
+    super(apiUrl, senderAddress, signer, gasPrice, gasLimits, broadcastMode);
+    this.cw3ContractAddress = cw3ContractAddress;
+  }
+
+  public getThreshold(): Promise<ThresholdResult> {
+    return this.queryContractSmart(this.cw3ContractAddress, { threshold: {} });
+  }
+
+  public getProposal(proposalId: number): Promise<ProposalResult> {
+    return this.queryContractSmart(this.cw3ContractAddress, { proposal: { proposal_id: proposalId } });
+  }
+
+  public listProposals({ startAfter, limit }: StartAfterNumberPaginationOptions = {}): Promise<
+    ProposalsResult
+  > {
+    return this.queryContractSmart(this.cw3ContractAddress, {
+      list_proposals: {
+        start_after: startAfter,
+        limit: limit,
+      },
+    });
+  }
+
+  public reverseProposals({ startBefore, limit }: StartBeforeNumberPaginationOptions = {}): Promise<
+    ProposalsResult
+  > {
+    return this.queryContractSmart(this.cw3ContractAddress, {
+      reverse_proposals: {
+        start_before: startBefore,
+        limit: limit,
+      },
+    });
+  }
+
+  public getVote(proposalId: number, voter: string): Promise<VoteResult> {
+    return this.queryContractSmart(this.cw3ContractAddress, {
+      vote: {
+        proposal_id: proposalId,
+        voter: voter,
+      },
+    });
+  }
+
+  public listVotes(
+    proposalId: number,
+    { startAfter, limit }: StartAfterStringPaginationOptions = {},
+  ): Promise<VotesResult> {
+    return this.queryContractSmart(this.cw3ContractAddress, {
+      list_votes: {
+        proposal_id: proposalId,
+        start_after: startAfter,
+        limit: limit,
+      },
+    });
+  }
+
+  public getVoter(address: string): Promise<VoterResult> {
+    return this.queryContractSmart(this.cw3ContractAddress, {
+      voter: {
+        address: address,
+      },
+    });
+  }
+
+  public listVoters({ startAfter, limit }: StartAfterStringPaginationOptions = {}): Promise<VotersResult> {
+    return this.queryContractSmart(this.cw3ContractAddress, {
+      list_voters: {
+        start_after: startAfter,
+        limit: limit,
+      },
+    });
+  }
+
+  public createMultisigProposal(
+    title: string,
+    description: string,
+    msgs: ReadonlyArray<Record<string, unknown>>,
+    earliest?: Expiration,
+    latest?: Expiration,
+    memo = "",
+  ): Promise<ExecuteResult> {
+    const handleMsg = {
+      propose: {
+        title: title,
+        description: description,
+        msgs: msgs,
+        earliest: earliest,
+        latest: latest,
+      },
+    };
+    return this.execute(this.cw3ContractAddress, handleMsg, memo);
```

Comment:
This came to me as an idea for a future iteration. I guess we could/should wait til stargate to do this to avoid wasted work:

We could create a helper function to convert from CosmosSDK message types to CosmosMsg types. Thus, you could take a normal bank/send message (or transaction) and then convert it to the corresponding CosmosMsg types and embed it in a proposal. This would be a stepping stone for apps that were unaware of the mutlisig - they request you to sign a staking message, and the wallet would convert that to the proper CosmosMsg type and embed it in a proposal.

You could click on the normal UI without modifying the dApp and the wallet would just turn this into a multisig action. Of course this would fail for anything but bank, staking, and wasm calls, but that is still a big and important space to cover.

What do you think? If so, let's make an issue for it

Reply:
https://github.com/cosmos/cosmjs/issues/548